### DISCUSSION TITLE
fill select options from context depending on other select field

### DISCUSSION THREAD

**Discussion by OmarRGtz** [react-hook-form/react-hook-form#2508] [Q&A] [answered]
Your hooks are awesome! Its a bliss to work with react-hook-form; still at the moment i am totally unsure how to solve the following simple use case.

I need a select field that is filled with options from data in Context that comes from another API, so far so good, but then i need a second select field that is filled based on the first one. I am totally unsure how to proceed.
What i need is to do an onChange event on the first one and then... ? Sorry, seems i have yet to grasp the React fu right.

Anyway, this is a code sandbox:
https://codesandbox.io/s/fill-react-hook-dev-select-options-from-context-jwvtf?file=/src/State.js

I am quite sure this common use case will help me a lot and a lot of others, since its the typical "select country then select state" case. I also need that some "countries" display not a select field, but an input number field and i am also quite lost on that...

Many thanks.

**Comment by bluebill1049** [comment, marked as answer]
Take a look watch API: https://react-hook-form.com/api#watch i think you just want conditional render right?

**Reply by OmarRGtz** [reply]
Not really, i need to feel the options of the second select field according to the current selection of the first field. The usual select country then select state, that populates the second select based on the first.

**Reply by bluebill1049** [reply]
so you want to watch the first select right?

then populate depend on what's been select in the second select?

**Reply by OmarRGtz** [reply]
Yep, from data in a react context state.

**Reply by bluebill1049** [reply]
Is this what you after? 

https://codesandbox.io/s/nervous-chaum-3pr82?file=/src/App.js

**Reply by OmarRGtz** [reply]
Exactly that behavior, but that solution wont work for me.
The first select is going to be filled dynamically with data from an api that might change but at least has 10 options, so a hardcoded ternary operator is no good.
How would a classic "select country then state" would be implemented in hook-forms if the countries were fetched from an api? I can do the legwork, but i am not sure which tools to use.

**Reply by OmarRGtz** [reply]
I think i am being terrible at explaining, sorry has been a long day, basically I want to reproduce this funcionality https://codesandbox.io/embed/formik-example-3jfxh?fontsize=14&hidenavigation=1&theme=dark
The classic state and country, the caveat is that i want to do it on field pairs generated with useFieldArray, that part is already done and working beautifully, but for the love of all that is holy i cant figure out how to make the second select of my Field Array to get values depending on the value of the first.

**Reply by OmarRGtz** [reply]
If you have a hint, would appreciate it greatly, its really cool working with react-hook-forms.

**Reply by bluebill1049** [reply]
i don't see much the difference with what i supplied above, but here is an updated version: 
https://codesandbox.io/s/nervous-chaum-3pr82?file=/src/App.js

**Reply by yogeshwar-chaudhari-20** [reply]
This is great. How do we do this in useFieldArray()?
One input field array has 2 select input fields.
The API response will populate 1st select input field. The options for the second field will be populated depending on the selection of the first input field.
There can be `n` number of groups.

**Comment by OmarRGtz** [comment]
You are right, i was derping around. Will give this a go when a bit more sharp.
